packages/typespec-client-generator-core/src/http.ts

```diff
@@ -128,26 +133,26 @@ function getSdkHttpParameters(
   // we add correspondingMethodParams after we create the type, since we need the info on the type
   const correspondingMethodParams: SdkModelPropertyType[] = [];
   if (tspBody) {
-    // if there's a param on the body, we can just rely on getSdkHttpParameter
+    // explicit @body and @bodyRoot
     if (tspBody.property && !isNeverOrVoidType(tspBody.property.type)) {
-      const getParamResponse = diagnostics.pipe(
-        getSdkHttpParameter(context, tspBody.property, httpOperation.operation, "body")
-      );
-      if (getParamResponse.kind !== "body") {
+      const bodyParam = diagnostics.pipe(
+        getSdkHttpParameter(context, tspBody.property, httpOperation.operation, undefined, "body")
+      )!;
+      if (bodyParam.kind !== "body") {
         diagnostics.add(
           createDiagnostic({
             code: "unexpected-http-param-type",
             target: tspBody.property,
             format: {
               paramName: tspBody.property.name,
               expectedType: "body",
-              actualType: getParamResponse.kind,
+              actualType: bodyParam.kind,
             },
           })
         );
         return diagnostics.wrap(retval);
       }
-      retval.bodyParam = getParamResponse;
+      retval.bodyParam = bodyParam;
```

Comment:
What are we modifying here? I didn't see the relationship with uri template.

Reply:
just a naming refactor